#! /bin/sh
patch -p0 -f $* < $0
exit $?

2009-08-07  Werner Koch  <wk@g10code.com>

	* crlfetch.c (my_es_read): Add explicit check for EOF.

2009-07-20  Werner Koch  <wk@g10code.com>

	* ldap.c (end_cert_fetch_ldap): Release the reader.
	Might fix bug#999.



--- src/ldap.c
+++ src/ldap.c
@@ -1420,7 +1451,11 @@ end_cert_fetch_ldap (cert_fetch_context_t context)
 {
   if (context)
     {
+      ksba_reader_t reader = context->reader;
+
       xfree (context->tmpbuf);
       xfree (context);
+      ldap_wrapper_release_context (reader);
+      ksba_reader_release (reader);
     }
 }


--- src/crlfetch.c      (revision 313)
+++ src/crlfetch.c      (working copy)
@@ -105,6 +105,12 @@ my_es_read (void *opaque, char *buffer, size_t nby
   result = es_read (cb_ctx->fp, buffer, nbytes, nread);
   if (result)
     return result;
+  /* Fixme we should check whether the semantics of es_read are okay
+     and well defined.  I have some doubts.  */
+  if (nbytes && !*nread && es_feof (cb_ctx->fp))
+    return gpg_error (GPG_ERR_EOF);
+  if (!nread && es_ferror (cb_ctx->fp))
+    return gpg_error (GPG_ERR_EIO);
 
   if (!cb_ctx->checked && *nread)
     {




